compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -1188,4 +1236,46 @@ private boolean isWithinParenthesis(Token openParen, Token closeParen) {
         return !isParenthesisMissing(openParen, closeParen) &&
                 isPosWithinOpenCloseLineRanges(linePosition, openParen.lineRange(), closeParen.lineRange());
     }
+
+    private Optional<TypeSymbol> transformFunctionOrMethod(BLangNode bLangNode, BType variableType,
+                                                           BInvokableSymbol bLangOriginalLangLib,
```

Comment:
This is not always a langLib. Can be a method or even a function defined in the source. So shall we use something like `originalInvokable` as the parameter name?